### ISSUE TITLE
After upgrading to 1.8.2, the log reports warning about no module named `dbt.adapters.duckdb`

### ISSUE THREAD

**Issue by devcovato** [dagster-io/dagster#23952]
### Dagster version

dagster, version 1.8.2

### What's the issue?

After upgrading to 1.8.2 and executing `dagster dev`, the log reports a warning about missing module `dbt.adapters.duckdb`.  Below is a fragment log (redacted where necessary):

```shell
2024-08-27 09:56:06 +0000 - dagster.builtin - WARNING - An error was encountered when creating a handle to the dbt adapter in Dagster.
Traceback (most recent call last):
  File "/workspaces/<dagster-project-name>/.venv/lib/python3.12/site-packages/dagster_dbt/core/resource.py", line 380, in _initialize_adapter
    from dbt.adapters.duckdb.credentials import DuckDBCredentials
ModuleNotFoundError: No module named 'dbt.adapters.duckdb'
```

In this project `duckdb` is not used.

### What did you expect to happen?

Since `duckdb` is not used, the log should not report any warning about missing the module.

### Deployment type

Local

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by garethbrickman** [comment]
Line in the code this error is thrown from: https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-dbt/dagster_dbt/core/resource.py#L383

**Comment by tobiplay** [comment]
Installing `dbt-duckdb` (temporarily) fixes the issue (for those looking to just suppress the warning right now).

**Comment by kennycontreras** [comment]
👋  I'm in version `1.8.6` and the error persist. Should I open a new issue?

**Comment by garethbrickman** [comment]
Fixed in 1.8.7
